### ISSUE TITLE
Remove old Python solver

### ISSUE THREAD

**Issue by richterjakob** [StanfordCBCL/svZeroDPlus#28]
svZeroDPlus has its own Python interface. The old python solver is no longer needed and is drifting apart from the C++ version feature-wise anyway.  IMO, deleting the python files would make sense.

Please note: This does not apply to the `runnercpp.py` file as it is part of the interface between python and C++.

**Comment by menon-karthik** [comment]
This is part of ongoing discussions with @ktbolt with regards to moving svZeroDPlus to the SimVascular repo. We will act on this after @ktbolt has had a chance to look at the code in more detail so we can make a plan for what needs to be done to move this to SimVascular.

**Comment by mrp089** [comment]
I propose to remove the Python solver here **now** since
- the `C++` version is extremely well-tested and documented (kudos to @richterjakob!!)
- we merged a major update with `svzerodcalibrator` that's not part of the `Python` version
- the 0D-3D interface is not supported in `Python`
- we already have minor inconsistencies between `Python` and `C++` (as @ncdorn experienced)
- adding new 0D elements (the main contribution by other lab members) is similarly easy in `C++` than in `Python`

 After that, I would do the following for consistency:
- archive https://github.com/SimVascular/svZeroDSolver
- move this repository to https://github.com/SimVascular

@richterjakob can perform the surgery in the next couple of days. @menon-karthik, @ktbolt, please speak now or forever hold your peace 😊

**Comment by menon-karthik** [comment]
I'm all for this! But let's give @ktbolt a chance to look at the code first.

**Comment by ktbolt** [comment]
@menon-karthik I'll have a look at the code today.

**Comment by ktbolt** [comment]
I had a look at some of the code and documentation, comments below.

**Documentation Review**

The `README` documentation needs to be improved 
- stating what is created when building: library and two executables
- a section describing the 3D solver interface

I can do this.

The `C++ Documentation` https://stanfordcbcl.github.io/svZeroDPlus/cpp/ needs to be improved 
- discusses Python
- can be organized a bit better

I can do this.

The documentation describing governing equations, code functionality, JSON file format, Python API, etc. needs to be in the SimVascular web site [here](https://simvascular.github.io/documentation/rom_simulation.html#0d-solver) not in github.io.


`Developer Guide` documentation is just a short _Architecture_ paragraph. There needs to be a _Design_ section that describes the problem, its components (i.e., objects) and how it is solved. Define any terms that will be used to relate the classes created in the implementation. 

There needs to be an _Implementation_ section that describes how the concepts in the `Design` section are realized in code. Describe the system structure (classes) and behavior, showing how the primary classes interact. Have a section describing how a developer can add new functionality.

`Code Comments`  
- some code (bloodvesseljunction.hpp) document equations which are not unreadable and essentially meaningless when a developer is looking in the source file. This kind of thing should be in the documentation. 
- don't need to add comments that are totally obvious (@brief Destroy the Block object)
- have comments of class method declaration rather than implementation


**Code Review**
I have identified a lot of issues in the code that need to be addressed. 

The primary issue (so far) in the implementation is the use of templates to define class float or double precision. There is no strong use-case for this and it makes the code cluttered and more complicated. So the first change to make is to remove all of the class templates and just use double.

I will list the other issues here after I finish my review of the code.

**Comment by menon-karthik** [comment]
Thanks for reviewing the code and docs, @ktbolt! I created a new issue (#35) and milestone related to replacing Simvascular/svZeroDSolver with svZeroDPlus. Let's continue this discussion in #35. We can reference the new milestone when creating new issues for code and documentation updates that are aimed at moving svZeroDPlusover to Simvascular. 

Meanwhile, this issue started off with just removing the Python solver within svZeroDPlus, before I derailed the conversation in my very first comment. So this shouldn't affect Simvascular/svZeroDSolver and the python solver here is essentially just a redundant version of that (correct @richterjakob?). So it still makes sense to remove it from the current repository.
